timeboost/src/binaries/mkconfig.rs

```diff
@@ -44,12 +50,25 @@ enum Mode {
 }
 
 impl Args {
-    fn mk_node_infos(&self, dec: &TrustedKeyMaterial) -> impl Iterator<Item = Result<NodeInfo>> {
-        iter::repeat_with(multisig::Keypair::generate)
-            .zip(iter::repeat_with(|| x25519::Keypair::generate().unwrap()))
+    fn mk_node_infos(
+        &self,
+        dec: &TrustedKeyMaterial,
+        seed: Option<u64>,
+    ) -> impl Iterator<Item = Result<NodeInfo>> {
+        let num_nodes: u8 = self.num.into();
+        let mut s_rng = secp256k1::rand::rngs::StdRng::seed_from_u64(
+            seed.map(|s| s.wrapping_pow(2)).unwrap_or_else(rand::random),
+        );
+        let mut d_rng = secp256k1::rand::rngs::StdRng::seed_from_u64(
+            seed.map(|s| s.wrapping_pow(3)).unwrap_or_else(rand::random),
+        );
+        iter::repeat_with(move || multisig::Keypair::generate_with_rng(&mut s_rng))
+            .zip(iter::repeat_with(move || {
+                x25519::Keypair::generate_with_rng(&mut d_rng).unwrap()
+            }))
             .zip(&dec.2)
             .enumerate()
-            .take(dec.2.len())
+            .take(num_nodes as usize)
```

Comment:
The `TrustedKeyMaterial` contains as many shares as there are nodes and since this method already takes the shares as a parameter, why not use its length?